frontend/src/scenes/session-recordings/sessionRecordingLogic.ts

```diff
@@ -165,6 +195,29 @@ export const sessionRecordingLogic = kea<sessionRecordingLogicType>({
                 sessionPlayerData?.snapshots.length > 1 && // more than one snapshot needed to init rrweb Replayer
                 !!sessionPlayerData?.snapshots?.find((s: eventWithTime) => s.type === 2), // there's a full snapshot in the data that was loaded
         ],
+        eventsApiParams: [
+            (selectors) => [selectors.sessionPlayerData],
+            (sessionPlayerData) => {
+                if (
+                    !sessionPlayerData?.person?.id ||
+                    !sessionPlayerData?.session_recording?.start_time ||
+                    !sessionPlayerData?.session_recording?.recording_duration
+                ) {
+                    return null
+                }
+
+                const buffer_ms = clamp(sessionPlayerData.session_recording.recording_duration / 4, 0, 30000) // +- before and after start and end of a recording to query for.
```

Comment:
Why not just always buffer by a minute or so on both sides?

Reply:
Hmm that's fair. Making the buffer dependent on the lenght of the recording may be complicating things.